### DISCUSSION TITLE
Structural Variants of CYP2D6 with WES

### DISCUSSION THREAD

**Discussion by emeadd** [pypgx/pypgx#89] [Q&A] [answered]
Hello. I would like to use PyPGx to analyze WES VCFs to determine CYP2D6 diplotypes. If there is a case with a gene deletion (*5) what will be reported in the output file?  Thank you very much for your help.

**Comment by sbslee** [comment, marked as answer]
Hello @emeadd,

Thanks for your interest in PyPGx.

First of all, please note that PyPGx requires depth of coverage data from BAM files in addition to a VCF file for detecting structural variants. Therefore, if VCF is all you have, then I'm afraid SV detection is not possible.

Secondly, assuming you have access to BAM files and are able to generate depth of coverage data, you can use PyPGx to detect structural variants including CYP2D6*5 (gene deletion). The output includes diplotype calls (e.g. `CYP2D6*1/*5`), phenotype calls (e.g. `Intermediate Metabolizer`), and plots of [copy number and allele fraction profiles](https://pypgx.readthedocs.io/en/latest/readme.html#structural-variation-detection).

For more details, please visit the [Read the Docs](https://pypgx.readthedocs.io/en/latest/readme.html#). You may also be interested in doing a short tutorial to run the PyPGx [pipeline](https://pypgx.readthedocs.io/en/latest/tutorials.html#get-rm-wgs-tutorial).

**Comment by Pharmacogenetecist** [comment]
Please move to a separate section if necessary - This question is the "best//ideal/accurate" way to assess CYP2D6 and CYP2B6 from WES.

I have been working on WES data. I have 48 WES samples including NA12878 aligned to GRCh38.
My Steps:
1. Prepare vcf file of pgx variants from BAM files using: pypgx create-input-vcf
2. Prepare depth of coverage: pypgx prepare-depth-of-coverage, this is using the .bed file for WES probes.
3. Prepare control statistics: pypgx compute-control-statistics and specify VDR gene region chr12:47841537-47943048 and .bed file as above.
4. Run NGS pipeline with the --platform Targeted

I came across the --samples-without-sv flag, how important would this be when using the above workflow? Or how else would you recommend improving it for more accurate results?

**Reply by sbslee** [reply]
@Pharmacogenetecist,

Thanks for sharing your WES pipeline! That's exactly what I would suggest.

As for the `--samples-without-sv` option from the `run-ngs-pipeline` command, its significance depends on the number of samples you are running together. If you have a fairly large cohort, then you probably don't need to worry about it. However, if you are running a dozen of samples, then you should probably look into the option. That's because, for SV detection using targeted sequencing data (`--platform Targeted`), PyPGx performs inter-sample normalization of read depth to compute copy number. If `--samples-without-sv` is not provided, PyPGx uses the population mean as normalization factor, which works pretty well if you have a large number of samples (some may have SV but most of them won't, so the population mean comes out as a good proxy for what an ideal sample without SV would look like). However, when there are only a few samples, then this population-based mean is not as stable as before and could be influenced significantly if there are samples with SV. That's why in this case you probably want to explicitly indicate "control" samples or those without SV. PyPGx will then use only these samples as normalization factor.

Note that inter-sample normalization is specific to targeted sequencing data. WGS does not require this step. Hope this helps!

**Reply by Pharmacogenetecist** [reply]
Thanks @sbslee
